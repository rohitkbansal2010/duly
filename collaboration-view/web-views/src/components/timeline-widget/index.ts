export { TimelineWidget } from './timeline-widget';
