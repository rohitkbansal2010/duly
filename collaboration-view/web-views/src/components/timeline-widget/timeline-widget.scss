@import "~@styles/spacing-constants.scss";

.timelineWidgetContainer {
  display: flex;
  height: 100%;
  max-height: $widget-height;
}
