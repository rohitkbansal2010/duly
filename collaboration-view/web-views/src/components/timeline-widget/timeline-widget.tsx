import { TimelineAppointments } from '@components/timeline-appointments';
import { PatientAppointmentAlias } from '@enums';

import styles from './timeline-widget.scss';

export const TimelineWidget = () =>
  (
    <div className={styles.timelineWidgetContainer}>
      {Object.values(PatientAppointmentAlias).map(alias =>
        (
          <TimelineAppointments key={alias} alias={alias} />
        ))}
    </div>
  );
