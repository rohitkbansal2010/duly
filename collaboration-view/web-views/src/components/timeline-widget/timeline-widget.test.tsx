import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { PatientAppointmentAlias } from '@enums';

import { TimelineWidget } from './timeline-widget';

const countTimelineAppointmentsComponents = Object.values(PatientAppointmentAlias).length;

const aliases = [ PatientAppointmentAlias.RECENT, PatientAppointmentAlias.UPCOMING ];

describe('TimelineWidget', () => {
  let component: ReturnType<typeof shallow>;

  beforeEach(() => {
    component = shallow(<TimelineWidget />);
  });

  it('should render the TimelineWidget component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it(`should render ${countTimelineAppointmentsComponents} the TimelineAppointments components`, () => {
    const timelineAppointmentsComponents = component.find('TimelineAppointments');

    expect(timelineAppointmentsComponents).toHaveLength(countTimelineAppointmentsComponents);
    timelineAppointmentsComponents.forEach((timelineAppointmentComponent, index) => {
      const expectedProps = { alias: aliases[index] };

      expect(timelineAppointmentComponent.props()).toEqual(expectedProps);
    });
  });
});
