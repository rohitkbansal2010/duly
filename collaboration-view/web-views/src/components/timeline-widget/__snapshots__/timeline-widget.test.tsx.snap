// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineWidget should render the TimelineWidget component 1`] = `
<div
  className="timelineWidgetContainer"
>
  <TimelineAppointments
    alias="recentAppointments"
    key="recentAppointments"
  />
  <TimelineAppointments
    alias="upcomingAppointments"
    key="upcomingAppointments"
  />
</div>
`;
